src/lib/ngsi/ContextElementResponse.cpp

```diff
@@ -266,6 +268,20 @@ ContextElementResponse::ContextElementResponse
 
     contextElement.contextAttributeVector.push_back(caP);
   }
+
+  /* creDate and modDate as "virtual" attributes. The entityDoc.hasField(...) part is a safety meassure to prevent entities created with
+   * very old Orion version which didn't implement creation/modification date */
+  if (includeCreDate && entityDoc.hasField(ENT_CREATION_DATE))
+  {
+    ContextAttribute* caP = new ContextAttribute(DATE_CREATED, DATE_TYPE, (double) getIntOrLongFieldAsLong(entityDoc, ENT_CREATION_DATE));
+    contextElement.contextAttributeVector.push_back(caP);
+  }
+
+  if (includeModDate && entityDoc.hasField(ENT_MODIFICATION_DATE))
+  {
+    ContextAttribute* caP = new ContextAttribute(DATE_MODIFIED, DATE_TYPE, (double) getIntOrLongFieldAsLong(entityDoc, ENT_MODIFICATION_DATE));
+    contextElement.contextAttributeVector.push_back(caP);
+  }
 }
```

Comment:
I just found it a bit strange and wanted to raise a 'warning flag'.
If all is good, then ...

NTC